src/hashvalue.cc

```diff
@@ -3,20 +3,18 @@
 using namespace v8;
 
 NAN_METHOD(Hasher) {
-  NanScope();
+  unsigned int hash = ((int) info[0]->NumberValue() << 24)
+    | ((int) info[1]->NumberValue() << 16)
+    | ((int) info[2]->NumberValue() << 8)
+    | (int) info[3]->NumberValue();
```

Comment:
On a slightly related subject, is there a reason you don't use Int32Value() or Uint32Value() here?

I think you can even do this entirely in JS land, V8 should be able to lower it to pretty much the same code a C++ compiler would generate, with less overhead even.